extensions/amp-mowplayer/0.1/test/test-amp-mowplayer.js

```diff
@@ -40,38 +41,32 @@ describes.realWin(
       timer = Services.timerFor(win);
     });
 
-    function getMowPlayer(
-      attributes,
-      opt_responsive,
-      opt_beforeLayoutCallback
-    ) {
-      const mp = doc.createElement('amp-mowplayer');
-      for (const key in attributes) {
-        mp.setAttribute(key, attributes[key]);
-      }
-      mp.setAttribute('width', '250');
-      mp.setAttribute('height', '180');
+    function getMowPlayer(attributes, opt_responsive) {
+      const element = createElementWithAttributes(doc, 'amp-mowplayer', {
+        ...attributes,
+        width: 250,
+        height: 180,
+      });
       if (opt_responsive) {
-        mp.setAttribute('layout', 'responsive');
+        element.setAttribute('layout', 'responsive');
       }
-      doc.body.appendChild(mp);
-      return mp
+      doc.body.appendChild(element);
+      element.implementation_.baseURL_ =
+        // Use a blank page, since these tests don't require an actual page.
+        // hash # at the end so path is not affected by param concat
+        `http://localhost:${location.port}/test/fixtures/served/blank.html#`;
+      return element
         .build()
+        .then(() => element.layoutCallback())
         .then(() => {
-          if (opt_beforeLayoutCallback) {
-            opt_beforeLayoutCallback(mp);
-          }
-          return mp.layoutCallback();
-        })
-        .then(() => {
-          const mpIframe = mp.querySelector('iframe');
-          mp.implementation_.handleMowMessage_({
+          const iframe = element.querySelector('iframe');
+          element.implementation_.handleMowMessage_({
             origin: 'https://mowplayer.com',
-            source: mpIframe.contentWindow,
+            source: iframe.contentWindow,
             data: JSON.stringify({event: 'onReady'}),
           });
         })
-        .then(() => mp);
+        .then(() => element);
```

Comment:
nit: could we return element from the `then()` function above, since the `handleMowMessage_` doesn't seem to be a promise that we're waiting on?